src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -37,10 +37,7 @@
     public UnmodifiableHeaders(Headers headers) {
         var h = headers;
         var unmodHeaders = new Headers();
-        h.forEach((k, v) -> {
-                List<String> l = v == null ? List.of() : Collections.unmodifiableList(v);
-                unmodHeaders.put(k, l);
-        });
+        h.forEach((k, v) -> unmodHeaders.put(k, Collections.unmodifiableList(v)));
         this.map = Collections.unmodifiableMap(unmodHeaders);
         this.headers = h;
```

Comment:
Now that we have `unmodHeaders` we could simply keep it:

```
this.headers = unmodHeaders;
```

we no longer need to keep the original `headers` around.